databases/foundationdb/Portfile

```diff
@@ -0,0 +1,84 @@
+# -*- coding: utf-8; mode: tcl; tab-width: 4; indent-tabs-mode: nil; c-basic-offset: 4 -*- vim:fenc=utf-8:ft=tcl:et:sw=4:ts=4:sts=4
+
+PortSystem          1.0
+PortGroup           github 1.0
+PortGroup           cmake 1.1
+
+name                foundationdb
+version             6.3.12
+categories          databases
+license             Apache-2
+maintainers         {me.com:spam_brian @brianwells} openmaintainer
+platforms           darwin
+description         FoundationDB is a distributed database providing ACID transactions.
+long_description    \
+    FoundationDB is a scalable, fault-tolerant, ordered \
+    key-value store with full ACID transactions.  This port \
+    will install version ${version} of the FoundationDB \
+    binaries suitable for development on macOS.
+
+homepage            https://www.foundationdb.org/
+
+github.setup        apple foundationdb ${version}
```

Comment:
Sorry to nitpick even more, but if you are using `github.setup`, then you don't need to specify `name` and `version`.

`github.setup` allows you to replace those fields.

So you would start your Portfile with:

```
github.setup        apple foundationdb 6.3.12
```

...at line 7, after your PortGroup declarations, and if you do that, the `name` and `version` lines can be removed.

`github.setup` will populate `${name}` and `${version}` automatically.

Reply:
Thanks. I adjusted the Portfile to use the `name` and `version` set by `github.setup` instead of defining these seperately.

Reply:
Thanks.  2 more changes and then we'll run tests, and if it looks good, we'll commit.

1. Can you please add a `revision` field under `github.setup`. Revision is used when making changes to a port when it's version number has not actually change.  Incrementing it signals to MacPorts to rebuild the port to pick up changes (which usually only happens when the version changes).  Would look like so:
  ```
  github.setup        apple foundationdb 6.3.12
  reivison            0
  ```
   
2. Can we squash all commits to one?  As indicated in the checklist when you first opened this PR, MacPorts convention usually requests one commit per port update or submission.

Reply:
I've added `revision            0` and pushed again with all the commits squashed into one.

Reply:
Thank you @brianwells